controllers/os-ubuntu/pkg/generator/generator.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generator
+
+import (
+	"text/template"
+
+	template_gen "github.com/gardener/gardener-extensions/pkg/controller/operatingsystemconfig/oscommon/template"
+	"github.com/gobuffalo/packr/v2"
+)
+
+var cmd = "/usr/bin/cloud-init clean && /usr/bin/cloud-init --file %s init"
+
+//go:generate packr2
+
+// NewCloudInitGenerator creates a new Generator using the template file for ubuntu
+func NewCloudInitGenerator() (*template_gen.CloudInitGenerator, error) {
```

Comment:
Can you initialize this only once? It's always the same, so it should be fine to put into an `init` function.

Reply:
So you suggest removing the packet generator which is related to the specific OS controller and to make one common function 
func NewCloudInitGeneratorFromTemlateFile(templateFilePath, unitPath, cmd string) (*CloudInitGenerator, error) in /pkg/controller/operatingsystemconfig/oscommon/template/template_generator.go ?

Reply:
No, just put the content of this function into an `init` function, do `runtime.Must` on all errors instead of returning them and assign the result of `template_gen.NewCloudInitGenerator` to a global variable called `Generator`.

Reply:
@adracus 
```
var cloudInitGenerator *template_gen.CloudInitGenerator
func init() {
	box := packr.New("templates", "./templates")
	cloudInitTemplateString, err := box.FindString("cloud-init.template")
	runtime.Must(err)
	cloudInitTemplate, err := template.New("cloud-init").Parse(cloudInitTemplateString)
	runtime.Must(err)
	cloudInitGenerator = template_gen.NewCloudInitGenerator(cloudInitTemplate, template_gen.DefaultUnitsPath, cmd)
}
func CloudInitGenerator() *template_gen.CloudInitGenerator {
	return cloudInitGenerator
}
```
Do you mean this way?